<launch>
	<include file="$(find gazebo_ros)/launch/empty_world.launch">
        <arg name="world_name" value="$(find quicksilver_description)/world/auto_world.world" />
	</include>
	<param name="robot_description" command="$(find xacro)/xacro '$(find quicksilver_description)/urdf/quicksilver.urdf.xacro'" />
	<arg name="x" default="0.0" />
	<arg name="y" default="0.0" />
	<arg name="z" default="0.0" />
	<param name="rate" value="10" />
	<rosparam file="$(find quicksilver_description)/config/joint_velocity.yaml" command="load"/>
	<rosparam file="$(find quicksilver_description)/config/joint_velocity_controller.yaml" command="load"/> 
    <!-- Nodes -->
	<node name="quicksilver_node" pkg="quicksilver_description" type="controller.py" />
	<node name="odom_pub" pkg="quicksilver_description" type="odom_pub.py" />
	<!-- <node name="scan_trans" pkg="quicksilver_description" type="scan_dealer.py" /> -->
    <node name="urdf_spawner" pkg="gazebo_ros" type="spawn_model" output="screen" args="-urdf -x $(arg x) -y $(arg y) -z $(arg z) -model quicksilver -param robot_description"/>
	<node name="joint_state_publisher" pkg="joint_state_publisher" type="joint_state_publisher" output="screen" />
    <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
    <node name="controller_spawner" pkg="controller_manager" type="spawner" respawn="false" output="screen" ns="/quicksilver" 
	args="--namespace=/quicksilver
	joint_state_controller
	rim_wheel4_joint
	rim_wheel3_joint
	rim_wheel2_joint
	rim_wheel1_joint
	"/>
</launch>
